src/pages/index.js

```diff
@@ -2,46 +2,84 @@
 import Card from "../components/Card.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
+import { validatorConfig, cardsListSelector, apiData } from "../utils/constants.js";
 import "../pages/index.css";
-import { initialCards, validatorConfig, cardsListSection } from "../utils/constants.js";
 
 //select elements
 const profile = document.querySelector(".profile");
 const profileEditButton = profile.querySelector(".profile__edit-button");
 const profileAddButton = profile.querySelector(".profile__add-button");
-const profileName = profile.querySelector(".profile__title");
-const profileJob = profile.querySelector(".profile__description");
+const profileNameEl = profile.querySelector(".profile__title");
+const profileAboutEl = profile.querySelector(".profile__description");
+const profileAvatar = profile.querySelector(".profile__image");
+const profileAvatarEdit = profile.querySelector(".profile__image-edit");
 const profileModalForm = document.forms["profile-form"];
 const profileModalNameInput = profileModalForm.querySelector(".profile-modal__name-input");
 const profileModalDescInput = profileModalForm.querySelector(".profile-modal__desc-input");
 const addModalForm = document.forms["add-form"];
+const avatarEditForm = document.forms["avatar-edit-form"];
 
 //instantiate classes
-const userInfo = new UserInfo(profileName, profileJob);
+const userInfo = new UserInfo({
+  name: profileNameEl,
+  about: profileAboutEl,
+  avatar: profileAvatar,
+  apiHeaders: apiData.headers,
+  apiUser: apiData.currentUser,
+});
+userInfo.getUserInfo();
+
+const initialCardsApi = new Api({
```

Comment:
`new Api` should be called only **once** because all methods will be ready inside `Api`

According to the task you need to pass full `headers` and `baseUrl` into the constructor of `Api`. See here https://disk.yandex.ru/i/J93dbJp7eypXKQ . 


So, you'll have a `baseUrl` and inside `Api` you'll use it adding **endpoints** to it not to do that in `index.js`


All your methods will have clear understandable names